### DISCUSSION TITLE
Request ID tracing with envoy and custom plugin lib

### DISCUSSION THREAD

**Discussion by lucascanna** [mia-platform/community#576] [Q&A]
### What product are you having troubles with?

Console

### What Console version are you using?

13

### Description

Hello!

I am using the Envoy API Gateway along with a Node.js service built using Mia-Platform's Custom Plugin Lib (v6.0.2).

For the same incoming request, I noticed that the request ID in Envoy's logs is different from the one recorded in the service's logs. Additionally, for authenticated requests that pass through the authorization-service and client-credentials service, I observe two other distinct request IDs.

Do you have any insights on why the request ID is not consistent throughout the entire service chain?

Thanks!

**Comment by malta895** [comment]
Hi @lucascanna,
could you please check the contents of the env variables `ADDITIONAL_HEADERS_TO_PROXY` of the custom plugin lib services, and `HEADERS_TO_PROXY`  of the authn/authz services?
They should both contain the `request-id` header (not only the `x-request-id`)
